Common/util/path.cpp

```diff
@@ -173,6 +173,28 @@ String FixupSharedFilename(const String &filename)
     return fixed_name;
 }
 
+#if AGS_PLATFORM_OS_ANDROID
+    String GetPathInForeignAsset(const String &filename)
+    {
+        if(filename.IsNullOrSpace()) return filename;
+
+        if(filename[0] == '/')
+        {
+            String adjusted_filename = filename;
+            adjusted_filename.ClipLeft(1);
+            return adjusted_filename;
+        }
+        else if(filename[0] == '.' && filename[1] == '/')
+        {
+            String adjusted_filename = filename;
+            adjusted_filename.ClipLeft(2);
+            return adjusted_filename;
+        }
```

Comment:
The shorter code could be
```
if (filename[0] == '/' || (filename.GetLength() > 1 && filename[0] == '.' && filename[1] == '/))
{
    return filename.Section('/', 1, -1);
}
```

Reply:
Curiosity, is the length check needed? If the string is 0 terminated, then it has at least two characters if it's not null or empty. 🤔

Reply:
> Curiosity, is the length check needed? If the string is 0 terminated, then it has at least two characters if it's not null or empty.

That's a good point, the String does do regular test in operator[], but it has `assert` which will trigger in debug config.

Reply:
That is, right now the String class has this:
```
    inline char operator[](size_t index) const
    {
        assert(index < _len);
        return _cstr[index];
    }
```
May be changed to
```
    inline char operator[](size_t index) const
    {
        assert(index <= _len);
        return _cstr[index];
    }
```

IIRC String class guarantees _cstr is valid at all times, pointing to "" literal if was not initialized.

Reply:
Actually, this is not way too important, there are various ways to simplify it, one is to use same conditions but avoid creating local variables, example:
```
        if(filename[0] == '/')
        {
            return filename.Left(1);
        }
        else if(filename[0] == '.' && filename[1] == '/')
        {
            return filename.Left(2);
        }
```

Reply:
Actually, there are various ways to simplify this, one way to keep same conditions but only avoid creating local variables:
```
        if(filename[0] == '/')
        {
            return filename.Mid(1);
        }
        else if(filename[0] == '.' && filename[1] == '/')
        {
            return filename.Mid(2);
        }
```

Maybe it's the best one.
`Mid` acts like normal "substring".

Reply:
I went with this last suggestion! Pushed below!